modules/markdown/assets/js/jetpack-markdown-block.jsx

```diff
@@ -0,0 +1,101 @@
+/* global wp:true */
+/* global React:true */
+/* eslint-disable react/jsx-no-bind */
+/**
+ * External dependencies
+ */
+//const React = react;
+
+/**
+ * Internal dependencies
+ */
+const { __ } = wp.i18n;
+
+const {
+	registerBlockType,
+	PlainText,
+	BlockControls,
+} = wp.blocks;
+
+const el = wp.element.createElement;
+
+registerBlockType( 'jetpack/markdown-block', {
+
+	title: 'Markdown',
+
+	description: __( 'We are going to have MARKDOWN!.' ),
+
+	icon: el(
+		'svg',
+		{
+			xmlns: 'http://www.w3.org/2000/svg',
+			'class': 'dashicons',
+			width: '208',
+			height: '128',
+			viewBox: '0 0 208 128',
+			stroke: 'currentColor'
+		},
+		el(
+			'rect',
+			{
+				width: '198',
+				height: '118',
+				x: '5',
+				y: '5',
+				ry: '10',
+				'stroke-width': '10',
+				fill: 'none'
+			}
+		),
+		el(
+			'path', { d: 'M30 98v-68h20l20 25 20-25h20v68h-20v-39l-20 25-20-25v39zM155 98l-30-33h20v-35h20v35h20z' }
+		)
+	),
+
+	category: 'formatting',
+
+	attributes: {
+		content: {
+			type: 'string',
+			source: 'property',
+			selector: 'textarea',
+			property: 'textContent',
+		},
+	},
+
+	supports: {
+		html: false,
+	},
+	edit( { attributes, setAttributes, className, isSelected } ) {
+		return [
+			isSelected && (
+				<BlockControls key="controls">
+					<div className="components-toolbar">
+						<button
+							className={ 'components-tab-button is-active' }>
+							<span>Markdown</span>
+						</button>
+					</div>
+				</BlockControls>
+			),
+			<PlainText
+				className={ className }
+				markdown="1"
+				value={ attributes.content }
+				onChange={ ( content ) => setAttributes( { content } ) }
+				aria-label={ __( 'Markdown' ) }
+			/>,
+		];
+	},
+	save( { attributes, className } ) {
```

Comment:
Why are we saving a `PlainText` block here?

Reply:
Thank you for pointing this out. I think this was a combination of my inexperience with the API (I think I had started with [this](https://wordpress.org/gutenberg/handbook/blocks/introducing-attributes-and-editable-fields/) example) and getting unexpected results with different approaches.

It boiled down to the fact that using a textarea preserved the newlines in the input field faithfully on the POST. 

textarea
```html
<h1>Headline</h1> 
<p>This is a paragraph</p>
````

other node
```html
<h1>Headline This is a paragraph</h1
````

I left the PlainText block there for the result, not the functionality. This was a mistake. I can get the same effect and potentially avoid unnecessary processing by changing it to:

`<textarea className={ className }>{ attributes.content }</textarea>`

I welcome any feedback on this.

Reply:
> It boiled down to the fact that using a textarea preserved the newlines in the input field faithfully on the POST.

Does this still happen with `RawHTML`? Serialising a textarea doesn't seem like the right solution here.

I'm also worried about the use of a regex. From looking at eff948e, they don't seem easy to read, or maintain.

Reply:
I had tried `RawHTML` early on to no avail, but I did not spend a lot of time on that. I do agree that serializing the content of the text area only feels like the right direction. I will work on a solving that issue and thereby doing away with the regex that strips the `textarea`.

Reply:
The answer to this is a longer conversation but basically it lies in the fact that making an HTML node into a string takes into account whether or not that element recognizes `\n` newlines. So `texture`, `pre` will be turned into strings incorporating the newlines. Other wrappers will only preserve them if I pass the Markdown content in the `textContent` prop since Javascript recognizes `\n`. That made for even more cumbersome parsing since the newlines would disappear once it arrived at the Markdown plugin. I opted for `pre` since `textarea` is a [controlled component](https://reactjs.org/docs/forms.html#controlled-components).